### ISSUE TITLE
Reports landing page

### ISSUE THREAD

**Issue by damsleth** [Puzzlepart/did#476]
- [X] I have searched for existing issues to avoid creating duplicates

### TLDR
The Reports page should have a landing page with pre-fetch filtering capabilities, not just listing out "all events" as is now. 

### Description
Today, the Reports page list out every.single.event in the TimeEntries table. 
This scales very poorly, and the page takes longer to load for each week.  

The Reports page should have a UI to pre-filter reports, e.g. listing the Weeknumber/Month/Year/Employee filter together with a "Generate report"-button. Similar to the filter UI today, but horizontal with dropdowns.

**The main use case for the Reports page is fetching the current month's TimeEntries**, so there should buttons for a couple of predefined reports to generate, e.g.
* Current Month
* Previous Month
* Current Q / Year(?)

Ties in with the Azure Table Storage performance issue #154
